### PR TITLE
WRKLDS-1449: bump(k8s): update k8s.io/* dependencies to v1.31.1

### PR DESCRIPTION
**Pull request by atiratree**

### CODE REVIEW

**Comment by atiratree** [comment]
https://github.com/openshift/cluster-kube-scheduler-operator/pull/551 needs to be merged first to resolve the library-go bump

File: pkg/cmd/recoverycontroller/cmd.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ func (o *Options) Run(ctx context.Context) error {
 	}
 
 	resourceSyncController := resourcesynccontroller.NewResourceSyncController(
+		"KubeSchedulerOperator",
```
Comment (atiratree) [inline pkg/cmd/recoverycontroller/cmd.go:93]: same issue here: https://github.com/openshift/cluster-kube-apiserver-operator/pull/1722#discussion_r1777595279

Comment (atiratree) [reply pkg/cmd/recoverycontroller/cmd.go]: seems that we go with the operand name: https://github.com/openshift/cluster-kube-apiserver-operator/commit/6f3ed3bc9b991818a5203c387bed69c32387fd0b

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Pull request merged**